src/app/view/sharing/SharingProvider.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useReducer, ReactNode, Dispatch, useEffect } from 'react'
+
+import {
+  SharingState,
+  SharingAction,
+  SharingIntentStatus,
+  SharingActionType
+} from '/app/domain/sharing/models/SharingState'
+import { handleReceivedFiles } from '/app/domain/sharing/services/SharingData'
+import { handleSharing } from '/app/domain/sharing/services/SharingStatus'
+
+const assertNever = (action: never): never => {
+  throw new Error('Unexpected object', action)
+}
+
+export const sharingReducer = (
+  state: SharingState,
+  action: SharingAction
+): SharingState => {
+  switch (action.type) {
+    case SharingActionType.SetIntentStatus:
+      return { ...state, sharingIntentStatus: action.payload }
+    case SharingActionType.SetFilesToUpload:
+      return { ...state, filesToUpload: action.payload }
+    default:
+      return assertNever(action)
+  }
+}
+
+const SharingStateContext = React.createContext<SharingState | undefined>(
+  undefined
+)
+const SharingDispatchContext = React.createContext<
+  Dispatch<SharingAction> | undefined
+>(undefined)
+
+interface SharingProviderProps {
+  children: ReactNode
+}
+
+export const SharingProvider = ({
+  children
+}: SharingProviderProps): JSX.Element => {
+  const [state, dispatch] = useReducer(sharingReducer, {
+    sharingIntentStatus: SharingIntentStatus.Undetermined,
+    filesToUpload: []
+  })
+
+  useEffect(() => {
+    const cleanupSharingIntent = handleSharing(
+      (status: SharingIntentStatus) => {
+        dispatch({ type: SharingActionType.SetIntentStatus, payload: status })
+      }
+    )
+
+    handleReceivedFiles(files => {
```

Comment:
updated, but at this time the library only provides `ReceiveSharingIntent.clearReceivedFiles()`, and I'm not sure it's good enough. It might be needed to handle this ourselves in native code